django_saml2_auth/views.py

```diff
@@ -117,11 +117,23 @@ def denied(r):
     return render(r, 'django_saml2_auth/denied.html')
 
 
-def _create_new_user(username, email, firstname, lastname):
-    user = user_model.objects.create_user(username, email)
-    user.first_name = firstname
-    user.last_name = lastname
-    groups = [Group.objects.get(name=x) for x in settings.SAML2_AUTH.get('NEW_USER_PROFILE', {}).get('USER_GROUPS', [])]
+def _create_new_user(user_identity):
+    attributes_map = settings.SAML2_AUTH.get('ATTRIBUTES_MAP', {})
+    user_name = user_identity[settings.SAML2_AUTH.get(
+        'ATTRIBUTES_MAP', {}).get('username', 'UserName')][0]
```

Comment:
use the `attributes_map` variable here rather than going through settings again.